pkg/compose/compose.go

```diff
@@ -130,9 +130,13 @@ func (s *composeService) projectFromName(containers Containers, projectName stri
 		serviceLabel := c.Labels[api.ServiceLabel]
 		_, ok := set[serviceLabel]
 		if !ok {
+			serviceImage := c.Image
+			if serviceNameFromLabel, ok := c.Labels[api.ImageNameLabel]; ok {
+				serviceImage = serviceNameFromLabel
+			}
```

Comment:
Nit: this is what tripped me up originally; I think a comment explaining behavior is warranted, i.e. that the label will be _present_ but empty for locally-built images and why we intentionally want that behavior 🙃